master/buildbot/reporters/mail.py

```diff
@@ -43,6 +43,11 @@
 from buildbot.reporters.message import MessageFormatter as DefaultMessageFormatter
 from buildbot.util import service
 
+if PY3:
```

Comment:
One other aspect of the code is that StringIO is used to interact with sys.stdin.
On Python 2, sys.stdin deals with bytes, and on Python 3, sys.stdin deals with unicode.

So the conditional statements I implemented are needed for the code to work properly on Python 2 and Python 3.